### PR TITLE
refactor(images-to-azure-storage-account): use bicep instead of arm t…

### PR DESCRIPTION
**Pull request by mattias-fjellstrom**
.

### CODE REVIEW

File: images-to-azure-storage-account/get-policy.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<policies>
```
Comment (mattias-kindborg-at-work) [inline images-to-azure-storage-account/get-policy.xml:1]: Why do we need this policy when it's also inline in the Bicep template?

File: images-to-azure-storage-account/post-policy.xml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+<policies>
```
Comment (mattias-kindborg-at-work) [inline images-to-azure-storage-account/post-policy.xml:1]: Why do we need this policy when it's also inline in the Bicep template?

File: images-to-azure-storage-account/README.md [hunk 3]
```diff
@@ -23,22 +23,23 @@ In this example we create an application that sends images from an Axis camera t
...
+- Azure CLI ([install](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli)) with Azure Bicep extension installed (`az bicep install`)
 
 ## File structure
 
 ```
 images-to-azure-storage-account
-├── azuredeploy.json - Azure Resource Manager (ARM) template describing the Azure resources
-└── create-cloud-resources.sh - Deployment script for bash and Azure CLI
+├── create-cloud-resources.sh - Deployment script for bash and Azure CLI
+├── get-policy.xml - Azure API Management policy document
```
[the remark this answers is not in the document]
Comment (mattias-fjellstrom) [reply images-to-azure-storage-account/README.md]: Good idea, fixed it.

File: images-to-azure-storage-account/README.md [hunk 4]
```diff
@@ -54,7 +55,7 @@ cd acap-integration-examples-azure/images-to-azure-storage-account
 
 ### Deploy Azure resources
 
-Let's deploy the Azure resources receiving the images sent from a camera. The services are described in `azuredeploy.json` using a [Azure Resource Manager (ARM) template](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/overview). We have two alternatives when it comes to deploying this ARM template. The first alternative is to run a bash script that performs all the necessary commands. The second alternative is to run all the commands manually.
+Let's deploy the Azure resources receiving the images sent from a camera. The services are described in `main.bicep` using an [Azure Bicep template](https://docs.microsoft.com/en-us/azure/azure-resource-manager/bicep/). We have two alternatives when it comes to deploying the cloud infrastructure. The first alternative is to run a bash script that performs all the necessary commands. The second alternative is to run all the commands manually.
```
[the remark this answers is not in the document]
Comment (mattias-fjellstrom) [reply images-to-azure-storage-account/README.md]: I removed it

File: images-to-azure-storage-account/main.bicep [hunk 5]
```diff
@@ -0,0 +1,123 @@
+@description('E-mail address to receive system notifications sent from API Management.')
+param publisherEmail string
+
+var commonName = 'image-upload-${uniqueString(resourceGroup().id)}'
+var storageBlobDataContributorRoleId = 'ba92f5b4-2d11-453d-a403-e96b0029c9fe'
```
[the remark this answers is not in the document]
Comment (mattias-fjellstrom) [reply images-to-azure-storage-account/main.bicep:8]: Added a reference to the documentation.

File: images-to-azure-storage-account/main.bicep [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    name: 'default'
+
+    resource container 'containers' = {
+      name: 'images'
+    }
+  }
+}
+
+resource roleAssignment 'Microsoft.Authorization/roleAssignments@2020-08-01-preview' = {
+  name: guid(resourceGroup().id)
```
[the remark this answers is not in the document]
Comment (mattias-fjellstrom) [reply images-to-azure-storage-account/main.bicep:28]: The name must be a valid and unique GUID.

File: images-to-azure-storage-account/README.md [hunk 7]
```diff
@@ -69,17 +70,16 @@ The following output indicates that the resources have been created successfully
 ```
 $ ./create-cloud-resources.sh MyResourceGroup eastus example@example.com
```
[the remark this answers is not in the document]
Comment (mattias-fjellstrom) [reply images-to-azure-storage-account/README.md:72]: Added a note about this in the troubleshooting section.

File: images-to-azure-storage-account/README.md [hunk 8]
```diff
@@ -23,22 +23,23 @@ In this example we create an application that sends images from an Axis camera t
...
+The camera will send images to the blob container via an API Management endpoint. The API Management resource receives a POST request containing the image from the camera. This request is transformed to a PUT request as this is required by the Azure storage REST API. The API Management uses a system-assigned managed identity to authenticate to Azure storage. The API Management endpoint is secured by a subscription key which is provided to the camera.
 
 ## Prerequisites
 
 - A network camera from Axis Communications (example has been verified to work on a camera with firmware version >=9.80.3.1)
-- Azure CLI ([install](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli))
+- Azure CLI ([install](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli)) with Azure Bicep extension installed (`az bicep install`)
```
[the remark this answers is not in the document]
Comment (mattias-fjellstrom) [reply images-to-azure-storage-account/README.md]: Good idea, I separated it in two lines.

File: images-to-azure-storage-account/create-cloud-resources.sh [hunk 9]
```diff
@@ -28,38 +28,20 @@ az group create \
   --location $location \
   --output none
 
-echo "Deploying ARM template..."
```
[the remark this answers is not in the document]
Comment (mattias-fjellstrom) [reply images-to-azure-storage-account/create-cloud-resources.sh:31]: This limits who can deploy this template (i.e. you need to have at least `Microsoft.Resources/deployments/validate/action` on the subscription scope). I tested deploying the template with limited rights (contributor on a single resource group, but not the deployment) and it works as expected, but not if I try to deploy the resource group using Bicep. Are we ok with this limitation?

[the remark this answers is not in the document]
Comment (mattias-kindborg-at-work) [reply images-to-azure-storage-account/create-cloud-resources.sh:31]: Can we create the resource group as a step in the documentation then, before the Bicep template is deployed? That way we get cross-platform support at least?

[the remark this answers is not in the document]
Comment (mattias-fjellstrom) [reply images-to-azure-storage-account/create-cloud-resources.sh:31]: Ok. I removed the deployment script and the documentation part where it was used. The whole process now consists of two commands, one to create a resource group and one to deploy the template and fetch the required output in one go.

**Review by mattias-kindborg-at-work — APPROVED** [review]

**Pull request merged**